app/controllers/idv/document_capture_controller.rb

```diff
@@ -17,7 +18,9 @@ def show
       Funnel::DocAuth::RegisterStep.new(current_user.id, sp_session[:issuer]).
         call('document_capture', :view, true)
 
-      render :show, locals: extra_view_variables
+      if !rate_limit_redirect!(:idv_doc_auth)
```

Comment:
Wondering if it'd be clearer / more conventional to call a `before_action` to do the redirect logic, so it doesn't need to be considered as part of the action itself (i.e. the action would only be reached if nothing else redirected).

Reply:
It's not the pattern we've been using for these post-FSM controllers, but it might be good to shift in that direction. The analytics wouldn't be logged in that case, and it would have to be only :show. The other thing I was thinking was to make the before_action in IdvStepConcern cover the show case, and just delegate to the controller for update to check if the last attempt succeeded.